plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -622,39 +655,30 @@ private function get_hpos_setting_for_sync() {
 				if ( ! $is_dangerous ) {
 					$sync_message[] = wp_kses_data(
 						sprintf(
-						// translators: %d: number of pending orders.
+							// translators: %s: number of pending orders.
 							_n(
-								"There's %d order pending sync. You can switch order data storage <strong>only when the posts and orders tables are in sync</strong>.",
-								'There are %d orders pending sync. You can switch order data storage <strong>only when the posts and orders tables are in sync</strong>.',
+								"You can switch order data storage <strong>only when the posts and orders tables are in sync</strong>. There's currently %s order out of sync.",
+								'You can switch order data storage <strong>only when the posts and orders tables are in sync</strong>. There are currently %s orders out of sync. ',
```

Comment:
>You can switch order data storage <strong>only when the posts and orders tables are in sync</strong>.

We can extract out the common message here too, but I'll leave it upto you to do it part of this PR or another one.